### DISCUSSION TITLE
Feature request: Option to add a file for download e.g pdf

### DISCUSSION THREAD

**Discussion by garethworld** [gearsdigital/enhanced-toolbar-link-dialog#86] [General]
It would be really useful to be able to add a file to create a download link.

**Comment by gearsdigital** [comment]
That is actually a really good idea. I'll have look on that!

Thank your very much for reaching out @garethworld

**Comment by gearsdigital** [comment]
@garethworld I had some time to let this sink in a bit. While I _really_ like the idea, I'm uncertain that this makes sense from a usability perspective:
1. Some of the current interface elements aren't useful for files (e.g. anchor). What should I do? Ignore? Disable?
2. I need probably new interface elements so that a user can decide if he want to add a download attribute.
3. The download attribute has some restrictins itself. Chrome 65+ and Firefox supporting only same-origin download links.
5. How do I distinguish between Pages and Files in the dropdown?

One thing that came into my mind, was to provide a "File" Dialog in the Writer Toolbar:

![toolbar](https://user-images.githubusercontent.com/965069/184853463-ccb52731-c01a-4478-8d2c-3cebb6efbe32.png)

This aligns not only with the Kirby default textareas, where "Files" is a sperated icon (and Dialog) but provides also a clear responsibility. Furthermore a User doesn't need to learn a new interface. Anyhow, extending the Writer-Toolbar is possible but not in an easy way 🙊

There exists a Plugin called [oh-hi-mark](https://github.com/rasteiner/oh-hi-mark) by @rasteiner where he solved this issue but to be honest, I really fear the maintence effort if something changes in the core.

I could extend this or create an additional plugin which is part of _enhanced-toolbar-link-dialog_. I could also provide a new one wich is based on _oh-hi-mark_. But than you have to download, install and maintain an additional plugin (probably two)...

So it is complicated...  But maybe I overthink this a bit. 😂

**Comment by garethworld** [comment]
I think you are spot on with a separate button, much cleaner and easier to understand.

**Comment by rasteiner** [comment]
my 2 cents: "oh-hi-mark" was intended only to add to a discussion. 
Its a hack that was less hacky than the plugin it was based on: https://github.com/johannschopplich/kirby-writer-marks which at the time required the user to patch the kirby core. 

Johann later rewrote his plugin and made it even less hacky than mine, so if you're considering to use one, it should probably be his.

**Comment by gearsdigital** [comment]
Thanks @rasteiner!

@garethworld Yes, it is. Still not sure about if (and how) I'm going to implement this. I'll let you know.

**Comment by purplespider** [comment]
This would be my ideal Kirby link dialog:

<img src="https://user-images.githubusercontent.com/329880/203524056-160cae46-ec63-4f00-ac25-3232dd1ea84c.png" width="502">

The fields would change based on the selected "Link type".

_(Mockup based on the link dialog from the Markdown plugin, where it alrady has the radio buttons for "type". I have added the "Page" (plus dropdown) and "File" options)._
